### PR TITLE
Add parameterized ObjectQueue tests

### PR DESCRIPTION
**Pull request by f2prateek**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -45,10 +45,11 @@
     <dagger.version>0.9</dagger.version>
     <otto.version>1.3.1</otto.version>
     <gson.version>2.2</gson.version>
-    <junit.version>4.10</junit.version>
+    <junit.version>4.11</junit.version>
```
Comment (f2prateek) [inline pom.xml]: Might be worth documenting this requirement on burst - I kept running into the error
`java.lang.NoSuchMethodError: org.junit.runner.Description.createTestDescription(Ljava/lang/String;Ljava/lang/String;[Ljava/lang/annotation/Annotation;)Lorg/junit/runner/Description;`

File: tape/src/test/java/com/squareup/tape/ObjectQueueTest.java [hunk 2]
```diff
@@ -14,110 +15,103 @@
...
+    };
+
+    public abstract <T> ObjectQueue<T> create(File file, FileObjectQueue.Converter converter) throws
+        IOException;
+  }
+
   @Rule public TemporaryFolder folder = new TemporaryFolder();
-  private FileObjectQueue<Pojo> queue;
+  @Burst QueueFactory factory;
+  ObjectQueue<String> queue;
```
Comment (f2prateek) [inline tape/src/test/java/com/squareup/tape/ObjectQueueTest.java]: Should we keep using a `Pojo` type here? Didn't see big upside/downside of doing so.

Comment (swankjesse) [inline tape/src/test/java/com/squareup/tape/ObjectQueueTest.java]: I like this.

File: pom.xml (hunk 1 above)
Comment (JakeWharton) [inline pom.xml]: It should be reflected in its dependencies. We happen to be overriding it with an old version which Maven does wrong.

File: tape/src/test/java/com/squareup/tape/ObjectQueueTest.java (hunk 2 above)
Comment (JakeWharton) [inline tape/src/test/java/com/squareup/tape/ObjectQueueTest.java]: Agreed. This is better.

File: tape/src/test/java/com/squareup/tape/ObjectQueueTest.java [hunk 3]
```diff
@@ -182,6 +176,9 @@ public void toStream(Pojo o, OutputStream bytes) throws IOException {
       iterator.next();
       fail();
     } catch (NoSuchElementException ignored) {
+      // thrown by file implementation.
+    } catch (IndexOutOfBoundsException ignored) {
+      // thrown by memory implementation.
```
Comment (JakeWharton) [inline tape/src/test/java/com/squareup/tape/ObjectQueueTest.java]: Can we normalize this?

Comment (f2prateek) [inline tape/src/test/java/com/squareup/tape/ObjectQueueTest.java]: yup!

**Pull request merged**